iceoryx_posh/source/roudi/port_pool.cpp

```diff
@@ -91,17 +92,19 @@ PortPool::addRunnableData(const iox::cxx::CString100& process,
 
 void PortPool::removeInterfacePort(popo::InterfacePortData* const portData) noexcept
 {
-    m_portPoolDataBase->m_interfacePortMembers.erase(portData);
+    m_portPoolDataBase->m_interfacePortMembers.remove_if(
+        [&](popo::InterfacePortData& port) { return &port == portData; });
 }
 
 void PortPool::removeApplicationPort(popo::ApplicationPortData* const portData) noexcept
 {
-    m_portPoolDataBase->m_applicationPortMembers.erase(portData);
+    m_portPoolDataBase->m_applicationPortMembers.remove_if(
+        [&](popo::ApplicationPortData& port) { return &port == portData; });
```

Comment:
Please do not capture everything. In this case we only require to capture the `portData` pointer so please change the lambda to `[portData](...`